Add Proceeding Model For State Courts

MorganBennetDev opened the pull request:
## Summary
<!-- What does this fix, how did you fix it, what approach did you take, what gotchas are there in your code or compromises did you make? -->
This PR adds a `Proceeding` model to the `search` module, in order to better support state court data going forward. Most state court data can be captured using existing models (with the caveat that using `RECAPDocument` for documents not sourced from RECAP may cause some confusion). However, existing models appear inadequate to handle dockets that may include an arbitrary number of hearings with a wide variety of types, as seen for example in the San Francisco Superior Court's data. Going forward, this should allow comprehensive capture of data from any state-level court so more changes should not be required. I was not able to find any major downsides to this approach aside from increased database usage.

## Deployment

**This PR should:**
<!-- The following labels control the deployment of this PR if they’re applied. -->
<!-- Please put an "X" in the box on ones that apply. -->
<!-- For more details on what pods are affected by each label, see the wiki -->
<!-- https://github.com/freelawproject/courtlistener/wiki/Pull-requests-%60skip%E2%80%90%7Btype%7D%E2%80%90deploy%60-labels -->

<!-- Check here if the entire deployment can be skipped -->
<!-- This might be the case for a small fix, a tweak to documentation or something like that. -->
- [ ] `skip-deploy` (skips everything below)
    <!-- Check here if the web tier can be skipped -->
    <!-- This is the case if you're working on code that doesn't affect the front end, like management commands, tasks, or documentation. -->
    - [ ] `skip-web-deploy`
    <!-- Check here if the deployment to celery can be skipped -->
    <!--This is the case if you make no changes to tasks.py or the code that tasks rely on. -->
    - [x] `skip-celery-deploy`
    <!-- check this if deployment to cron jobs can be skipped -->
    <!-- This is the case if no changes are made that affect cronjobs. -->
    - [ ] `skip-cronjob-deploy`
    <!-- Deployment of daemons can be skipped -->
    <!-- This is the case if you haven't updated daemons or the code they depend on. -->
    - [ ] `skip-daemon-deploy`

MorganBennetDev committed:
feat(search): add Proceeding model

Allows a wider variety of court proceedings to be tracked in order to support state court dockets.

MorganBennetDev committed:
Merge branch 'freelawproject:main' into main

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/freelawproject/courtlistener?pullRequest=6401) <br/>All committers have signed the CLA.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

MorganBennetDev committed:
feat(search): add pg.track decorator to Proceeding

Implements change tracking for the new Proceeding model with the pg.track decorator.

MorganBennetDev committed:
Merge branch 'freelawproject:main' into main

mlissner commented:
@albertisfu what should we do with this one? Full review or punt for later?

MorganBennetDev committed:
Merge branch 'freelawproject:main' into main
